### PR TITLE
Add security options for OpenApi

### LINKED ISSUE #1082: Add securityDefinitions to openapi/swagger spec
Hi,

it would be nice if we could include a securityDefinitions field to the swagger.json. By doing this we can have a custom header field in swagger-ui where you can paste your jwt.

right now I can't set an Authorization Bearer token with each request. but swagger-ui would show a button if some securityDefinitions is included in the openapi/swagger json spec I get from the root path.

or is there already any other way to do this?

### PR DESCRIPTION
**Pull request by laurenceisla**
Closes #1082.

- [x] Support security options
- [x] Refactor code
- [x] Add configuration option
- [x] Add changelog

### CODE REVIEW

**Comment by laurenceisla** [comment]
Tested for Swagger and the options `securityDefinitions` and `security` are enough for the `Authorize` button to appear and handle authorized requests. Although this applies to all the endpoints, I don't see an easy way for now to differentiate between an endpoint that needs authorization and one that doesn't.

**Comment by steve-chavez** [comment]
> I don't see an easy way for now to differentiate between an endpoint that needs authorization and one that doesn't.

Hm, let's just make it configurable for now.

```
openapi-security-definitions = ".."
```

disabled by default.

File: test/io/configs/expected/no-defaults-with-db-other-authenticator.config [hunk 1]
```diff
@@ -19,6 +19,7 @@ jwt-secret = "ODERREALLYREALLYREALLYREALLYVERYSAFE"
 jwt-secret-is-base64 = true
 log-level = "info"
 openapi-mode = "disabled"
+openapi-security-active = true
```
Comment (steve-chavez) [inline test/io/configs/expected/no-defaults-with-db-other-authenticator.config:22]: Hm, I think this one required changing the in-db config(like below). Would be good to test this too.

https://github.com/PostgREST/postgrest/blob/bdf1cbe11126bae043b754009a1a7fe19f9c7e9f/test/io/db_config.sql#L56-L57

**Review by steve-chavez — APPROVED** [review]

**Comment by steve-chavez** [comment]
Hm, somehow a test failed on main branch https://github.com/PostgREST/postgrest/runs/7333244906?check_suite_focus=true

**Comment by laurenceisla** [comment]
> Hm, somehow a test failed on main branch https://github.com/PostgREST/postgrest/runs/7333244906?check_suite_focus=true

It passed now.

**Pull request merged**